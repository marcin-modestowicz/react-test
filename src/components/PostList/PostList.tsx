import React from 'react';
import Post from 'components/Post';

const PostList = () => <div><Post/></div>;

export default PostList;
