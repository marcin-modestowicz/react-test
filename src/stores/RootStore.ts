import PostStore from './PostStore';
import UserStore from './UserStore';

class RootStore {
    postStore: PostStore;
    userStore: UserStore;

    constructor() {
        this.postStore = new PostStore(this);
        this.userStore = new UserStore(this);
    }
}

export default RootStore;
