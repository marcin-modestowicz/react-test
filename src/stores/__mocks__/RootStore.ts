import PostStore from '../PostStore';
import UserStore from '../UserStore';
import Post from 'models/Post';
import User from 'models/User';

class RootStore {
    postStore: PostStore;
    userStore: UserStore;

    constructor() {
        this.postStore = new PostStore(this);
        this.userStore = new UserStore(this);

        this.postStore.posts.push(new Post({ id: 1, userId: 1, title: 'title', body: 'content' }));
        this.userStore.users.set(1, new User({ id: 1, name: 'John Doe' }))
    }
}

export default RootStore;
