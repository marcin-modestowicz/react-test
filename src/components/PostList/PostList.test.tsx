import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import RootStore from "stores/RootStore";
import PostList from "./PostList";

jest.mock("stores/RootStore");

describe("PostList component", () => {
  test("should render", () => {
    const postList = shallow(
        // @ts-ignore
        <PostList.wrappedComponent rootStore={new RootStore()}/>
    );
    expect(toJson(postList)).toMatchSnapshot();
  });
});
