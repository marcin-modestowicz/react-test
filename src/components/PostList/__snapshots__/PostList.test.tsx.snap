// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostList component should render 1`] = `
<section>
  <Post
    author="John Doe"
    content="content"
    key="1"
    title="title"
  />
</section>
`;
